메인화면 리팩토링 phase2

yys7517 opened the pull request:
## 💡 관련 이슈
<!--close #이슈넘버-->
close #55

## 🌱 변경 사항 및 이유
<!--변경사항 적기-->
- 메인 어댑터 람다식, 리스너 인터페이스로 변경
- 신고, 차단 다이얼로그 Anchor View로 보여지게 수정
- 메인화면 탑 버튼
- 스크롤에 따른 메인화면 탑 버튼 visibility 설정
- 신고, 차단 팝업 showReportSelectLeft
- 신고, 차단 팝업 showReportSelectBottom
- 중복 클릭 방지를 위한 Flow Binding 적용
- 메인화면 스크롤 상태 유지
- 카테고리 태그 버튼 크기 조절
- 인기 Teen 유저 아이템 크기 조정
- 인기 Teen 자동 스크롤

Commit:
신고, 차단 선택 다이얼로그 border 제거
#55

Commit:
신고, 차단 선택 다이얼로그 Anchor View 형식으로 보여주는 함수 추가
#55

Commit:
메인 어댑터 람다식, 리스너로 변경
#55

Commit:
인기 Teen 리스트, 이전 유저 스크롤 이슈 수정
#55

yys7517 committed:
중복클릭 방지를 위한 Flow Binding Clicks 확장함수
#55

yys7517 committed:
신고, 차단 다이얼로그 디바이더 추가
#55

yys7517 committed:
신고, 차단 다이얼로그 Bottom 방향 앵커뷰 추가
#55

yys7517 committed:
메인화면 탑 버튼, 스크롤 유지
#55

yys7517 committed:
메인화면 카테고리 태그 버튼 크기 수정 #55

yys7517 committed:
메인화면 툴바 수정, 상단 이동 버튼 visibility 수정 #55

yys7517 committed:
인기 Teen 리스트 유저 아이템 크기 조절 #55

yys7517 committed:
Merge branch 'refs/heads/develop' into yys/refactor_main_#55

yys7517 committed:
인기 Teen 리스트 자동 스크롤 #55

kim0hoon commented on app/src/main/java/com/eighteen/eighteenandroid/presentation/home/MainFragment.kt:
```diff
@@ -99,6 +175,148 @@ class MainFragment : BaseFragment<FragmentMainBinding>(FragmentMainBinding::infl
+                handler.postDelayed(autoScrollRunnable!!, 4500)
+            }
+
+            override fun stopAutoScroll() {
```
onDestroyView 에서도 자동 스크롤을 stop 시켜줘야 하지 않나요??
아니면 코루틴(viewLifecycleOwner.lifecycleScope)으로 구현하셨어도 무방하셨을 것 같습니다!

yys7517 replied:
앗.. 놓치고 있었군요. 감사합니다!

kim0hoon replied:
오 벌써 반영해주셨네요 ㅎㅎㅎ
뷰 관련 작업이 있어 lifecyclescope -> viewLifecycleOwner.lifecycleScope를 사용하셔야 할 것 같아요
그리고 viewLifecycleOwner.lifecycleScope를 사용하신다면 onDestroyView와 onDestory에서 따로 cancel시켜주실 필요는 없습니다~

yys7517 replied:
자꾸 놓치는 부분들 체크해주셔서 감사합니다! Fragment에서는 viewLifeCycleOwner 사용하도록 꼭 생각해야겠네요

yys7517 committed:
자동 스크롤 Coroutine으로 수정 및 동작 cancel 코드 추가 #55

yys7517 committed:
viewLifeCycleOwner의 lifecycleScope 사용하도록 수정 #55

Commit:
Teen 아이템 수정 #55
- 인기 Teen width, height
- Teen, 인기 Teen 그라데이션 수정

kim0hoon committed:
코드 품질 향상 #55
- 인기 Teen width, height
- Teen, 인기 Teen 그라데이션 수정
- 다른 아이템 보일 때, 인기 Teen 자동 스크롤 멈춤

Signed-off-by: yys <yys>
